src/jvm/main/org/jetbrains/kotlinx/lincheck/util/Logger.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.util
+
+import java.io.*
+import java.text.SimpleDateFormat
+import java.util.*
+
+object LincheckLogger {
+    private val LINE_SEPARATOR = System.lineSeparator()
+    private val formatter: ThreadLocal<SimpleDateFormat> = object : ThreadLocal<SimpleDateFormat>() {
+        override fun initialValue(): SimpleDateFormat {
+            return SimpleDateFormat("dd.MM.yyyy HH:mm:ss")
+        }
+    }
+
+    private val logFilename = System.getProperty("lincheck.log.file") ?: null
+    private val logFile = if (logFilename != null) File(logFilename) else null
+    private var logWriter: Writer? = null
+
```

Comment:
Let's also re-use `LogLevel` enum from `Reporter.kt`  (and let's move it into this file), and use it as a parameter to set logging level.